### ISSUE TITLE
Nondescriptive error when upgrading a package which has been removed: "Package was not found in rpm database"

### ISSUE THREAD

**Issue by ppisar** [rpm-software-management/dnf5#492]
When running "dfn5 upgrade", I noticed an unneeded package. So I run "dnf remove" for that single package in parallel. The removal finished before a transaction check of the upgrade and naturally the transaction check of the upgrade failed because of the package became missing:

~~~~
[163/163] Total                                                                                                                                                           100% |  13.7 MiB/s | 279.7 MiB |  00m20s
Verifying PGP signatures

Running transaction
Package was not found in rpm database
~~~~

My complain is that the error message does name what package is missing. In my case it was texlive-lib package. I would expect:

~~~
texlive-lib package was not found in rpm database.
~~~

Observed with dnf5-5.0.7-20230405004417.109.g1a26c95b.fc38.x86_64.

**Comment by j-mracek** [comment]
I agree, we should check rpm cookie to identify whether RPMdb is still at the same state. I think it could be related enhancement of dnf5 lock.

**Comment by m-blaha** [comment]
Upstream Fedora bug report: https://bugzilla.redhat.com/show_bug.cgi?id=2350483

**Comment by evan-goode** [comment]
We now wait for the current running transaction to complete before resolving a new transaction: https://github.com/rpm-software-management/dnf5/pull/2519. But the error message may be obtainable by other means.
